File: 김채린/Stack/기능 개발.kt
import java.util.*

fun solution(
    progresses: IntArray,
    speeds: IntArray,
): IntArray {
    var index = 0
    var result = MutableList(0) { 0 }
    var stack = Stack<Int>()
    while (index < progresses.size) {
        var count = 0
        var n = (100 - progresses[index]) / speeds[index]
        if ((100 - progresses[index]) % speeds[index] != 0)
            {
                n++
            }
        stack.push(n)
        count++
        index++
        while (index < progresses.size) {
            val num = stack.pop()
            if (progresses[index] + (speeds[index] * num) >= 100)
                {
                    count++
                    index++
                    stack.push(num)
                } else {
                break
            }
        }
        result.add(count)
    }
    return result.toIntArray()
}

File: 정민선/프로그래머스/기능개발.py
def solution(progresses, speeds):
    answer = []
    days = []
    
    for p, s in zip(progresses, speeds):
        remain = 100 - p
        time = (remain + s) // s
        days.append(time)
        
    while days:
        day = days.pop(0)
        count = 1
        
        while days and days[0] <= day:
            days.pop(0)
            count += 1
        
        answer.append(count)
    
    return answer

progresses = [95, 90, 99, 99, 80, 99]
speeds = [1, 1, 1, 1, 1, 1]
print(solution(progresses, speeds))